src/udev/dmi_memory_id/dmi_memory_id.c

```diff
@@ -0,0 +1,808 @@
+                log_error("Entry point length too large (%u bytes, expected %u).",
+                          (unsigned)buf[0x05], 0x1FU);
+                return 0;
+        }
+
+        if (!verify_checksum(buf, buf[0x05])
+            || memcmp(buf + 0x10, "_DMI_", 5) != 0
+            || !verify_checksum(buf + 0x10, 0x0F))
+                return 0;
+
+        dmi_table(DWORD(buf + 0x18), WORD(buf + 0x16), WORD(buf + 0x1C),
+                  devmem, false, no_file_offset);
+
+        return 1;
+}
+
+static int legacy_decode(uint8_t *buf, const char *devmem, bool no_file_offset) {
+        if (!verify_checksum(buf, 0x0F))
+                return 0;
+
+        dmi_table(DWORD(buf + 0x08), WORD(buf + 0x06), WORD(buf + 0x0C),
+                  devmem, false, no_file_offset);
+
+        return 1;
+}
+
+static int parse_argv(int argc, char * const *argv) {
+        static const struct option options[] = {
+                { "help", no_argument, NULL, 'h' },
+                { "from-dump", required_argument, NULL, 'F' },
+                { "version", no_argument, NULL, 'V' },
+                {}
+        };
+
+        for (;;) {
+                int option;
+
+                option = getopt_long(argc, argv, "F:hV", options, NULL);
+                if (option == -1)
+                        break;
+
+                switch (option) {
+                case 'F':
+                        arg_source_file = optarg;
+                        break;
+                case 'h':
+                        printf("Usage: %s [options]\n"
+                               " -F,--from-dump FILE   read DMI information from a binary file\n"
+                               " -h,--help             print this help text\n\n",
+                               program_invocation_short_name);
+                        return 0;
+                case 'V':
+                        printf("%s\n", GIT_VERSION);
+                        return 0;
+                case '?':
+                        return -EINVAL;
+                }
+        }
+
+        return 0;
+}
+
+static int run(int argc, char* const* argv) {
+        int r = 0;
+        int found = 0;
+        size_t size;
+        _cleanup_free_ uint8_t *buf = NULL;
+        bool no_file_offset = false;
+
+        log_set_target(LOG_TARGET_AUTO);
+        udev_parse_config();
+        log_parse_environment();
+        log_open();
+
+        r = parse_argv(argc, argv);
+        if (r < 0)
+                return r;
+
+        /* Read from dump if so instructed */
+        r = read_full_file_full(AT_FDCWD,
+                                arg_source_file ?: SYS_ENTRY_FILE,
+                                0, 0x20, 0, NULL, (char **) &buf, &size);
+        if (r < 0) {
+                return log_full_errno(!arg_source_file && r == -ENOENT ? LOG_DEBUG : LOG_ERR,
+                                      r, "Reading \"%s\" failed: %m",
+                                      arg_source_file ?: SYS_ENTRY_FILE);
+        }
```

Comment:
coding style: we usually don't place {} around single-line if blocks

Reply:
Is it a single-line block if it uses multiple lines? :)

Reply:
Made the change so I can mark this as fixed.

Reply:
> Is it a single-line block if it uses multiple lines? :)

All that matters is that's it's a single *logical* line. The number of source lines doesn't matter ;)